src/app/my-projects/my-projects-edit.component.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { MyProjectsEditComponent } from './my-projects-edit.component';
+import { AppMaterialModule } from '../app-material.module';
+import { ReactiveFormsModule } from '@angular/forms';
+import { MAT_BOTTOM_SHEET_DATA, MatBottomSheetRef } from '@angular/material';
+import { MyProjectsService } from './my-projects.service';
+import { of } from 'rxjs';
+import { UserProject } from '../user-projects/user-project';
+import { Project } from '../projects/project';
+import { GlobalErrorHandlerService } from '../error/global-error-handler.service';
+import { MatMomentDateModule } from '@angular/material-moment-adapter';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+import * as moment from 'moment';
+import { Util } from '../util/util';
+
+describe('MyProjectsEditComponent', () => {
+  let component: MyProjectsEditComponent;
+  let fixture: ComponentFixture<MyProjectsEditComponent>;
+
+  const userProject: UserProject = {
+    id: 1,
+    role: 'developer',
+    tasks: 'development',
+    startDate: Util.ignoreTimezone(moment('2019-01-10')),
+    endDate: Util.ignoreTimezone(moment('2019-05-10')),
+    creationDate: moment(),
+    lastModifiedDate: moment(),
+    user: {
+      id: '123',
+      userName: 'username'
+    },
+    project: {
+      id: '456',
+      name: 'Project',
+      creationDate: new Date(),
+      customer: 'Customer',
+      description: null,
+      industrySector: 'Software development',
+      lastModifiedDate: new Date()
+    }
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ MyProjectsEditComponent ],
+      imports: [ AppMaterialModule, ReactiveFormsModule, MatMomentDateModule, BrowserAnimationsModule ],
+      providers: [
+        { provide: MatBottomSheetRef, useValue: jasmine.createSpyObj('matBottomSheetRef', ['dismiss'] ) },
+        { provide: MAT_BOTTOM_SHEET_DATA, useValue: userProject },
+        { provide: MyProjectsService, useValue: jasmine.createSpyObj('myProjectsService', {
+            'updateCurrentUserProject': of<UserProject>(null)
+          }) },
+        GlobalErrorHandlerService
+      ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MyProjectsEditComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should send a request to update user project', () => {
+    component.updateUserProject();
+    const myProjectService: MyProjectsService = TestBed.get(MyProjectsService);
+
+    expect(myProjectService.updateCurrentUserProject).toHaveBeenCalledWith('456', {
+      role: 'developer',
+      tasks: 'development',
+      startDate: Util.ignoreTimezone(moment('2019-01-10')),
+      endDate: Util.ignoreTimezone(moment('2019-05-10'))
+    });
+  });
+
```

Comment:
could you please add one more test which checks that the form was prefilled correctly with userProject data

Reply:
Thanks. Done.